source/virtualBuffers/gecko_ia2.py

```diff
@@ -167,9 +167,13 @@ def _normalizeControlField(self, attrs):  # noqa: C901
 			attrs["roleTextBraille"] = roleTextBraille
 		if attrs.get("IAccessible2::attribute_dropeffect", "none") != "none":
 			states.add(controlTypes.State.DROPTARGET)
-		if role == controlTypes.Role.LINK and controlTypes.State.LINKED not in states:
-			# This is a named link destination, not a link which can be activated. The user doesn't care about these.
-			role = controlTypes.Role.TEXTFRAME
+		if role == controlTypes.Role.LINK:
+			if controlTypes.State.LINKED not in states:
+				# This is a named link destination, not a link which can be activated. The user doesn't care about these.
+				role = controlTypes.Role.TEXTFRAME
+			elif config.conf["documentFormatting"]["reportLinkType"]:
+				if self.NVDAObjectAtStart.isInternalLink:
+					states.add(controlTypes.State.INTERNAL_LINK)
```

Comment:
Something like this, I think

```suggestion
			elif config.conf["documentFormatting"]["reportLinkType"]:
				if (value := attrs.get("IAccessible::value")) is not None) and urlUtils._valueToSamePage(value, self.obj.documentConstantIdentifier):
					states.add(controlTypes.State.INTERNAL_LINK)
```